### PR TITLE
Virt ffa

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Patches needed by https://patchew.org/Xen/20240205154909.2392231-1-jens.wiklander@linaro.org/

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
In commit message for "core: thread_spmc.c: add set_simple_ret_val()":
"**~~Simplifies~~ Simplify** common FFA_ERRROR/..."
"... is return**ed** on error or ..."

In commit message for "core: ffa: add SPMC_CORE_SEL1_MAX_SHM_COUNT":
```patch
-Adds SPMC_CORE_SEL1_MAX_SHM_COUNT, telling how many shared memory object
+Add SPMC_CORE_SEL1_MAX_SHM_COUNT, telling how many shared memory objects
```
(move `s` from first to last word :)

In commit message for "core: virt, ffa: keep guest partition until resources are reclaimed": s/Moves/Move/

Regarding the implementation: 
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for what it's worth.

File: core/arch/arm/kernel/virtualization.c [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 #include <kernel/panic.h>
 #include <kernel/refcount.h>
 #include <kernel/spinlock.h>
+#include <kernel/thread_spmc.h>
 #include <kernel/virtualization.h>
 #include <mm/core_memprot.h>
 #include <mm/core_mmu.h>
 #include <mm/tee_mm.h>
 #include <platform_config.h>
+#include <platform_config.h>
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/kernel/virtualization.c]: Thanks

**Comment by jenswi-linaro** [comment]
Comments addressed and tag applied.

**Comment by jforissier** [comment]
IBART failed, rebase needed?

**Comment by jenswi-linaro** [comment]
I've pushed the update, but it seems github doesn't notice it. Perhaps it's a temporary issue.

**Pull request merged**